tests/unit/testConfigs/bucketNotifConfigTest.js

```diff
@@ -0,0 +1,142 @@
+const assert = require('assert');
+const { bucketNotifAssert } = require('../../../lib/Config');
+
+describe('bucketNotifAssert', () => {
+    it('should not throw an error if bucket notification config is valid', () => {
+        assert.doesNotThrow(() => {
+            bucketNotifAssert([{
+                resource: 'target1',
+                type: 'kafka',
+                host: 'localhost',
+                port: 8000,
+                auth: { user: 'user', password: 'password' },
+            }]);
+        },
+        'should not throw an error if there is no bucket notification config');
+    });
+    it('should throw an error if bucket notification config is not an array', () => {
+        assert.throws(() => {
+            bucketNotifAssert({
+                resource: 'target1',
+                type: 'kafka',
+                host: 'localhost',
+                port: 8000,
+                auth: { user: 'user', password: 'password' },
+            });
+        },
+        '/bad config: bucket notification configuration must be an array/');
```

Comment:
this is the part where using asserts over if's kinda sucks. Using regex to match now requires us to keep error messages in sync with test code. That is, updating an error message will break a test. This, to me anyway, is not ideal as I should be able to update "flavor" text without breaking a test. Consider if a page header change broke a test, it doesn't really add any value for that. 

The alternative of using `if` is we can throw predefined errors and type match on those.

Anyway, that is my rant. I would say just asserting that it throws and ignoring the error message is probably fine but this is more of a team thing so up to everyone.

Reply:
This test format was basically copy/pasted from other config unit tests. I agree that it's not super practical to use error messages in tests, but it's the only way we have to differentiate where in the code the error is thrown. Only checking for an error leaves the possibility of there being an unintentional error in a test case or something like that. I'm defo into figuring out some better way of doing it, but I think the error messages need to be included.

Reply:
One method would be to turn the error messages into constants and compare that way. Comparing a thrown error against the expected const error ( as a string or error object ) would be fine imo. But I will approve either way as I am very late to the party.